app/model/commands/ActiveAssetCommand.scala

```diff
@@ -49,4 +57,43 @@ case class ActiveAssetCommand(
       AssetVersionConflict
     }
   }
+
+  private [commands] def firstFrameImageName(mp4Name: String): String = {
+    // drop .mp4 and replace with image suffix
+    mp4Name.dropRight(4).concat(VideoAsset.firstFrameImageSuffix)
+  }
+
+  /**
+   * use a default image generated from the first frame of video if:
+   *  - the new asset version to activate is self-hosted
+   *  AND
+   *  - a first-frame image is available
+   *  AND (
+   *    - the current posterImage is empty
+   *    OR
+   *    - the current posterImage is using a default image
+   *  )
+   */
```

Comment:
Clever. Some implicit logic going on here for the end user but I think the right amount that we don't need to communicate it to them.